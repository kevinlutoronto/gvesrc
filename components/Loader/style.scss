@import '../../styles/base';

.container {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: auto;
}

.spinner {
  margin: auto 0;
  text-align: center;
}

.spinner > div {
  width: 8px;
  height: 8px;
  background-color: #fff;
  border-radius: 100%;
  display: inline-block;
  margin: 0 2px;
  -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
  animation: sk-bouncedelay 1.4s infinite ease-in-out both;
}

.spinner .bounce1 {
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
  background-color: $cPurple200;
}

.spinner .bounce2 {
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
  background-color: $cPurple200;
}

.spinner .bounce3 {
  background-color: $cPurple200;
}

@-webkit-keyframes sk-bouncedelay {
  0%,
  80%,
  100% {
    -webkit-transform: scale(0);
  }
  40% {
    -webkit-transform: scale(1);
  }
}

@keyframes sk-bouncedelay {
  0%,
  80%,
  100% {
    -webkit-transform: scale(0);
    transform: scale(0);
  }
  40% {
    -webkit-transform: scale(1);
    transform: scale(1);
  }
}
