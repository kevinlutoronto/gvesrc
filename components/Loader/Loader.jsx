// @flow
import React from 'react';
import cn from 'classnames';
import style from './style.scss';
import { getDataAttrs } from '../../dataUtils';
import type { Data, StyleAttributes } from '../../types';

type Props = {|
  className?: string,
  dotColor?: string,
  dotClassName?: string,
  style?: StyleAttributes,
  dotStyle?: StyleAttributes,
  data?: Data,
|};

class Loader extends React.Component<Props> {
  static defaultProps = {
    dotStyle: {},
  };

  render() {
    const { dotColor, dotStyle = {} } = this.props;
    if (dotColor != null) {
      dotStyle.backgroundColor = dotColor;
    }

    return (
      <div
        {...getDataAttrs(this.props.data)}
        className={cn(style.container, this.props.className)}
        style={this.props.style}
      >
        <div className={style.spinner}>
          <div className={cn(style.bounce2, this.props.dotClassName)} style={dotStyle} />
          <div className={cn(style.bounce3, this.props.dotClassName)} style={dotStyle} />
          <div className={cn(style.bounce1, this.props.dotClassName)} style={dotStyle} />
        </div>
      </div>
    );
  }
}

Loader.displayName = 'Plasma@Loader';

export default Loader;
