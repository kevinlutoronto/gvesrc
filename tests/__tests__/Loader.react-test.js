/* eslint-disable */

import React from 'react';
import Loader from '../../components/Loader/Loader';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Loader Component', () => {
  test('render', () => {
    const component = renderer.create(<Loader />);
    expect(component).toMatchSnapshot();
  });

  test('sets dotColor as the dots background-color', () => {
    const component = renderer.create(<Loader dotColor="#c1c1c1" />);

    expect(component).toMatchSnapshot();
  });
});
