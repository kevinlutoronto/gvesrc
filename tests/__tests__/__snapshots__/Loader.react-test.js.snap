// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component render 1`] = `
<div
  className="container"
  style={undefined}
>
  <div
    className="spinner"
  >
    <div
      className="bounce2"
      style={Object {}}
    />
    <div
      className="bounce3"
      style={Object {}}
    />
    <div
      className="bounce1"
      style={Object {}}
    />
  </div>
</div>
`;

exports[`Loader Component sets dotColor as the dots background-color 1`] = `
<div
  className="container"
  style={undefined}
>
  <div
    className="spinner"
  >
    <div
      className="bounce2"
      style={
        Object {
          "backgroundColor": "#c1c1c1",
        }
      }
    />
    <div
      className="bounce3"
      style={
        Object {
          "backgroundColor": "#c1c1c1",
        }
      }
    />
    <div
      className="bounce1"
      style={
        Object {
          "backgroundColor": "#c1c1c1",
        }
      }
    />
  </div>
</div>
`;
